lib/github-pages-health-check/error.rb

```diff
@@ -1,13 +1,54 @@
 module GitHubPages
   module HealthCheck
     class Error < StandardError
+      DOCUMENTATION_BASE = "https://help.github.com"
+      DOCUMENTATION_PATH = "/categories/github-pages-basics/"
+      LOCAL_ONLY = false # Error is only used when running locally
+
+      attr_reader :repository, :domain
+
+      def initialize(repository: nil, domain: nil)
+        super
+        @repository = repository
+        @domain     = domain
+      end
+
       def self.inherited(base)
         subclasses << base
       end
 
       def self.subclasses
         @subclasses ||= []
       end
+
+      def message
+        "Something's wrong with your GitHub Pages site."
+      end
+
+      def message_with_url
+        msg = message.gsub(/\s+/, " ").squeeze(" ").strip
+        msg << "." unless msg =~ /\.$/
+        [msg, more_info].join(" ")
```

Comment:
This is really confusing. Can you write a comment about what it should produce, i.e. provide an example?